examples/gaussian_process/utils.py

```diff
@@ -0,0 +1,79 @@
+'''
+Gaussian Process utils. Design inspired by GPflow.
+'''
+
+import numpy as np
+import tensorflow as tf
+import zhusuan as zs
+
+
+class RBFKernel:
+
+    def __init__(self, n_inp, name='rbf_kernel', dtype=tf.float32):
+        k_raw_scale = tf.get_variable('k_log_scale_{}'.format(name),
+                                      [n_inp], dtype,
+                                      initializer=tf.zeros_initializer())
+        self.k_scale = tf.nn.softplus(k_raw_scale)
+
+    def __call__(self, x, y):
+        '''
+        Return K(x, y), where x and y are possibly batched.
+        :param x: shape [n_x, n_covariates] or [n_batch, n_x, n_covariates]
+        :param y: shape [n_y, n_covariates] or [n_batch, n_y, n_covariates]
```

Comment:
This can be made better by assuming x, y to be of shape `[..., n_x/y, n_covariates]`. `...` represents any number of dimensions. And all if else on static shapes can be avoided.

Reply:
done